Add  org.desmume.DeSmuME

linkmauve opened the pull request:
DeSmuME is a [Nintendo DS](https://en.wikipedia.org/wiki/Nintendo_DS) [emulator](https://en.wikipedia.org/wiki/Emulator) written in C++ using GTK.

DeSmuME hasn’t seen a release [since 2015-04-15](http://desmume.org/2015/04/15/desmume-0-9-11-released/) but has been progressing a lot in master since then, hence the `git` source. The maintainer would prefer not to do a release ever again.

linkmauve commented on org.desmume.DeSmuME.appdata.xml:
```diff
@@ -0,0 +1,49 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
This file is temporary, once it is reviewed here I’ll push it upstream.

linkmauve commented on org.desmume.DeSmuME.appdata.xml:
```diff
@@ -0,0 +1,50 @@
+
+  <screenshots>
+    <screenshot type="default">
+      <image>https://linkmauve.fr/files/desmume-pokémon.png</image>
```
This screenshot of DeSmuME at its native resolution is smaller than the recommended size, but I’d like to keep it that way instead of padding it with useless blank pixels, will that be an issue?

bilelmoussaoui commented on org.desmume.DeSmuME.appdata.xml:
```diff
@@ -0,0 +1,50 @@
+
+  <releases>
+    <release version="0.9.11" date="2015-04-15"/>
+    <release version="0.9.10" date="2013-11-28"/>
+    <release version="0.9.9" date="2013-04-30"/>
+    <release version="0.9.8" date="2012-04-09"/>
+    <release version="0.9.7" date="2011-02-01"/>
+    <release version="0.9.6" date="2010-05-19"/>
+    <release version="0.9.5" date="2009-11-27"/>
+    <release version="0.9.4" date="2009-07-08"/>
+    <!-- There is no 0.9.3, don’t look for it. -->
+    <release version="0.9.2" date="2009-04-12"/>
+    <release version="0.9.1" date="2009-02-13"/>
+    <release version="0.9" date="2009-01-04"/>
+    <release version="0.8" date="2008-04-22"/>
+    <release version="0.7.3" date="2007-08-12"/>
+    <release version="0.7.2" date="2007-07-10"/>
+    <release version="0.7.1" date="2007-06-13"/>
+    <release version="0.7.0" date="2007-05-11"/>
+    <release version="0.6.0" date="2007-02-04"/>
+    <release version="0.5.0" date="2006-12-23"/>
```
```suggestion
```
There's no need to keep all the versions, especially if they don't a description

bilelmoussaoui commented on org.desmume.DeSmuME.json:
```diff
@@ -0,0 +1,68 @@
+    "command": "desmume",
+    "rename-desktop-file": "desmume.desktop",
+    "finish-args": [
+        "--filesystem=xdg-config/desmume:create",
```
Just use $XDG_DATA_DIRS upstream and remove this permission

linkmauve replied:
Why?  Doesn’t this bind-mount `~/.config/desmume` into the sandbox, thus letting the user configure it from the documented directory in all cases?

bilelmoussaoui replied:
Users can still configure it from `.var/app/$app-id/config`.

linkmauve replied:
This is much less user-friendly though. :/

If possible I’d like to keep `~/.config/desmume` as the user configuration directory, to avoid having to modify all of the documentation everywhere.

That’s also what [I did in poezio](https://github.com/flathub/io.poez.Poezio/blob/master/io.poez.Poezio.json#L7) (where it is arguably even more important, since users are *expected* to modify these files manually).

barthalion replied:
We made an exception for poezio as it's terminal app. Graphical apps should respect XDG_CONFIG_HOME.

linkmauve replied:
Alright, I’ve removed this line.

bilelmoussaoui commented on org.desmume.DeSmuME.json:
```diff
@@ -0,0 +1,68 @@
+    "runtime-version": "20.08",
+    "sdk": "org.freedesktop.Sdk",
+    "command": "desmume",
+    "rename-desktop-file": "desmume.desktop",
```
Ideally, this should be renamed upstream

bilelmoussaoui commented on org.desmume.DeSmuME.json:
```diff
@@ -0,0 +1,68 @@
+            ]
+        }
+    ],
+    "cleanup": [
```
Per conventions, cleanup should be between finish-args & modules

bilelmoussaoui commented on org.desmume.DeSmuME.json:
```diff
@@ -0,0 +1,68 @@
+                }
+            ]
+        },
+        {
+            "name": "appdata",
+            "buildsystem": "simple",
+            "build-commands": [
+                "install -Dm644 -t /app/share/appdata/ org.desmume.DeSmuME.appdata.xml"
+            ],
+            "sources": [
+                {
+                    "type": "file",
+                    "path": "org.desmume.DeSmuME.appdata.xml"
+                }
+            ]
+        }
```
I guess this will moved upstream?

linkmauve replied:
Indeed, this is just to make it work for now.

bilelmoussaoui commented on org.desmume.DeSmuME.appdata.xml:
```diff
@@ -0,0 +1,50 @@
+  <project_license>GPL-2.0-or-later</project_license>
+  <metadata_license>CC0-1.0</metadata_license>
+  <developer_name>DeSmuME Team</developer_name>
+  <icon type="remote">https://linkmauve.fr/files/desmume-logo.svg</icon>
```
if you're already shipping an icon with the app, there's no need to add one unless you want it as a fallback?

bilelmoussaoui commented on org.desmume.DeSmuME.appdata.xml:
```diff
@@ -0,0 +1,50 @@
+  <metadata_license>CC0-1.0</metadata_license>
+  <developer_name>DeSmuME Team</developer_name>
+  <icon type="remote">https://linkmauve.fr/files/desmume-logo.svg</icon>
+
```
Could use a `<launchable type="desktop-id">org.desmume.DeSmuME.desktop</launchable>` (hence the suggestion to rename the desktop file upstream)

bilelmoussaoui commented:
bot, build org.desmume.DeSmuME

LeandroStanger commented on org.desmume.DeSmuME.json:
```diff
@@ -0,0 +1,68 @@
+            "sources": [
+                {
+                    "type": "file",
+                    "path": "org.desmume.DeSmuME.appdata.xml"
```
New appdata files should use metainfo instead. https://www.freedesktop.org/software/appstream/docs/chap-Quickstart.html

linkmauve commented:
There, I’ve fixed all of your comments (except for the `~/.config` one) and pushed the metainfo file upstream, as well as fixed the ARM build failure!

bilelmoussaoui commented:
bot, build org.desmume.DeSmuME

linkmauve commented:
I’ve replaced `--socket=fallback-x11` with `--socket=x11` because SDL 1.2 requires X11 access atm. I’ll work upstream on moving off from this library.

After testing, every feature seems to work properly, the `GtkFileChooserNative` correctly uses the portals to open files, etc. :)

The only issues so far were with X11, and also that my dark variant theme isn’t respected. Do I need to do something special for that? I’ve configured it in `~/.config/gtk-3.0/settings.ini` on the host system.

barthalion commented:
Your theme needs to be available on Flathub – then flatpak will install it automatically.

barthalion commented:
Also it looks like there's something wrong with desktop file as it uses stock icon here. Probably `Icon=` should set to app id.

linkmauve commented:
There, I’ve removed the `XDG_CONFIG_HOME` override and renamed the icon upstream!

> Your theme needs to be available on Flathub – then flatpak will install it automatically.

It is Adwaita’s dark variant, do I need to do anything special for it?

barthalion commented:
bot, build org.desmume.DeSmuME

barthalion commented:
> There, I’ve removed the `XDG_CONFIG_HOME` override and renamed the icon upstream!
> 
> > Your theme needs to be available on Flathub – then flatpak will install it automatically.
> 
> It is Adwaita’s dark variant, do I need to do anything special for it?

Hm, I don't remember the details. I thought it should work just fine; https://github.com/flathub/flathub/issues/1023 is one of the issues reported in the past. Maybe you'll find some hints there.

barthalion commented:
Somehow it's still not exported on my system but I'm not sure why; appstream-compose finds SVG file and I see PNG files in app-info dir. Does it work for you?

linkmauve commented:
I’ve installed GNOME and indeed it doesn’t display the icon. I’ve then compared with the build of poezio, and the main difference is that I’m using `/app/share/pixmaps/` in DeSmuME and `/app/share/icons/hicolor/scalable/apps/` in poezio. I’ll try doing that shortly.

linkmauve commented:
There, it works!

barthalion commented:
bot, build org.desmume.DeSmuME

barthalion commented:
LGTM. @linkmauve Should we give someone beside you write access to Flathub repo?

linkmauve commented:
I’m the only one atm caring about Linux support in this emulator, but if someone else from the project comes and ask for it please just grant them access. :)

barthalion commented:
/merge
